e2e/tcproute_test.go

```diff
@@ -727,3 +727,129 @@ func TestMeshWithMultipleTCPRoutes(t *testing.T) {
+	t.Cleanup(func() {
+		err := theCloud.TcpRoutes().Delete(ctx, tcprID.Key)
+		if err != nil {
+			t.Logf("delete TCProute: %v", err)
+		}
+
+		for _, bs := range bss {
+			err = theCloud.BackendServices().Delete(ctx, bs.Key)
+			if err != nil {
+				t.Logf("delete backend service: %v", err)
+			}
+		}
+		for _, hc := range hcs {
+			err = theCloud.HealthChecks().Delete(ctx, hc.Key)
+			t.Logf("theCloud.HealthChecks().Delete(ctx, %s): %v", hc.Key, err)
+		}
+	})
+
+	for i := 0; i < len(bss); i++ {
+		checkGCEBackendService(t, ctx, theCloud, hcs[i], bss[i], 80)
+	}
+
+	negID, err := buildNEG(graphBuilder, resUniqueIdPart+"-neg", zone)
+	if err != nil {
+		t.Fatalf("buildNEG((_, %s, tcpr-deletion-test-zone) = (_, %v), want (_, nil)", resUniqueIdPart, err)
+	}
+
+	for i := 0; i < len(bss); i++ {
+		bss[i], err = buildBackendServiceWithNEG(graphBuilder, resUniqueIdPart+"-"+strconv.Itoa(i)+"-bs", hcs[i], negID)
+		if err != nil {
+			t.Fatalf("buildBackendServiceWithNEG(_, %s, %s, %s) = (_, %v), want (_, nil)", resUniqueIdPart, hcs[i], negID, err)
+		}
+	}
+
+	expectedActions = []exec.ActionMetadata{
+		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, negID)},
+		{Type: exec.ActionTypeUpdate, Name: actionName(exec.ActionTypeUpdate, bss[0])},
+		{Type: exec.ActionTypeUpdate, Name: actionName(exec.ActionTypeUpdate, bss[1])},
+		{Type: exec.ActionTypeUpdate, Name: actionName(exec.ActionTypeUpdate, bss[2])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[0])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[1])},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcs[2])},
+		{Type: exec.ActionTypeMeta, Name: eventName(tcprID)},
+	}
+
+	processGraphAndExpectActions(t, graphBuilder, expectedActions)
+	t.Cleanup(func() {
+		err = theCloud.NetworkEndpointGroups().Delete(ctx, negID.Key)
+		t.Logf("theCloud.NetworkEndpointGroups().Delete(ctx, %s): %v", negID.Key, err)
+	})
+
+	graphBuilder.Get(tcprID).SetState(rnode.NodeDoesNotExist)
+	for i := 0; i < len(bss); i++ {
+		graphBuilder.Get(bss[i]).SetState(rnode.NodeDoesNotExist)
+		graphBuilder.Get(hcs[i]).SetState(rnode.NodeDoesNotExist)
+	}
+	expectedActions = []exec.ActionMetadata{
+		{Type: exec.ActionTypeMeta, Name: eventName(negID)},
+		{Type: exec.ActionTypeDelete, Name: actionName(exec.ActionTypeDelete, bss[0])},
+		{Type: exec.ActionTypeDelete, Name: actionName(exec.ActionTypeDelete, bss[1])},
+		{Type: exec.ActionTypeDelete, Name: actionName(exec.ActionTypeDelete, bss[2])},
+		{Type: exec.ActionTypeDelete, Name: actionName(exec.ActionTypeDelete, hcs[0])},
+		{Type: exec.ActionTypeDelete, Name: actionName(exec.ActionTypeDelete, hcs[1])},
+		{Type: exec.ActionTypeDelete, Name: actionName(exec.ActionTypeDelete, hcs[2])},
+		{Type: exec.ActionTypeDelete, Name: actionName(exec.ActionTypeDelete, tcprID)},
+	}
+
+	processGraphAndExpectActions(t, graphBuilder, expectedActions)
+	_, err = theCloud.TcpRoutes().Get(ctx, tcprID.Key)
+	if err == nil {
```

Comment:
you need to check that error is equal googleAPIs does not exist. otherwise this might be a different error

Reply:
Done.

Reply:
did you pushed a new code? I don't see the change.